File: java/SubString/subarraySum/subarraySum.java
import java.util.*;

public class subarraySum {
    public int subarraySumFunc(int[] nums, int k) {
        int count = 0;
        int prefix = 0;

        Map<Integer, Integer> mp = new HashMap<>();

        mp.put(0, 1);

        for (int num : nums) {
            prefix += num;

            if (mp.containsKey(prefix - k)) {
                count += mp.get(prefix - k);
            }

            mp.put(prefix, mp.getOrDefault(prefix, 0) + 1);

        }

        return count;
    }
}

File: Python/SubString/subarraySum/subarraySum.py
def subarraySum(nums: list[int], k: int) -> int:
    count = 0
    prefix = 0
    mp: dict[int, int] = {}

    mp[0] = 1

    for num in nums:
        prefix += num

        if prefix - k in mp:
            count += mp.get(prefix - k, 0)

        mp[prefix] = mp.get(prefix, 0) + 1

    return count
